### PR TITLE
Generate master playlists for fragmented-mp4 media playlists

### PR DESCRIPTION
**Pull request by emranemran**
transcode/manifest: generate master fmp4 playlists and upload to object-store

### CODE REVIEW

File: transcode/transcode.go [hunk 1]
```diff
@@ -385,6 +406,17 @@ func getHlsTargetURL(tsr TranscodeSegmentRequest) (*url.URL, error) {
 	}
 }
 
+func getFmp4TargetURL(tsr TranscodeSegmentRequest) (*url.URL, error) {
+	if tsr.FragMp4TargetUrl != "" {
```
Comment (mjh1) [inline transcode/transcode.go:410]: nit: reverse the logic and do `return nil, nil` here

File: clients/manifest.go [hunk 2]
```diff
@@ -183,6 +183,55 @@ func GenerateAndUploadManifests(sourceManifest m3u8.MediaPlaylist, targetOSURL s
...
+		} else {
+			var resolutionA = transcodedStats[a].Width * transcodedStats[a].Height
+			var resolutionB = transcodedStats[b].Width * transcodedStats[b].Height
+
+			return resolutionA > resolutionB
+		}
+	})
+
+	// For each profile, add a new entry to the master manifest
+	for i, profile := range transcodedStats {
```
Comment (mjh1) [inline clients/manifest.go:206]: Do we already have this logic for standard HLS that we could reuse?

**Review by mjh1 — APPROVED** [review]

File: transcode/transcode.go (hunk 1 above)
Comment (emranemran) [reply transcode/transcode.go:410]: Will change it.

File: clients/manifest.go (hunk 2 above)
Comment (emranemran) [reply clients/manifest.go:206]: Yeah i just copied for now. i'll turn this into a function.

**Comment by emranemran** [comment]
@mjh1 I'm handling the edits in a follow-up PR i'm doing - just wanted to merge this as the new PR depends on this and it's easier for me to make the changes in the new one.

**Pull request merged**